oauth2_http/java/com/google/auth/oauth2/ServiceAccountJwtAccessCredentials.java

```diff
@@ -160,6 +176,10 @@ static ServiceAccountJwtAccessCredentials fromJson(Map<String, Object> json, URI
     String privateKeyPkcs8 = (String) json.get("private_key");
     String privateKeyId = (String) json.get("private_key_id");
     String quoataProjectId = (String) json.get("quota_project_id");
+    String rawUniverseDomain = (String) json.get("universe_domain");
+    String resolvedUniverseDomain =
+        (rawUniverseDomain == null) ? Credentials.GOOGLE_DEFAULT_UNIVERSE : rawUniverseDomain;
```

Comment:
qq, is this for backwards compatibility? If the `universe_domain` is null, then we set it to GDU?

Should we also check if it's null or empty, we throw an exception?

Reply:
That's right, this defaults to GDU if it's empty. But good point, it might be better to throw an exception.

Reply:
After offline discussion, when reading from credential files, we want to default to GDU if a universe_domain is not provided. This is to support legacy credential file patterns.